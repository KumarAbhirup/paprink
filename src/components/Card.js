import React, { Component } from 'react'

export default class Card extends Component {
  render() {
    const { post } = this.props
    return (
      <div className={`card ${ this.props.type === 'small_image' && 'card_small_with_image' } ${ this.props.type === 'small_background' && 'card_default card_small_with_background' } ${ this.props.type === 'largest' && 'card_largest_with_image' } ${ this.props.type === 'mini' && 'card_default card_default_no_image' } grid-item`}>
        { this.props.type === 'small_background' && <div className="card_background" style={{backgroundImage:`url(${post.thumbnail})`}}></div> }
        { this.props.type != 'small_background' && this.props.type != 'mini' && <img className="card-img-top" src={post.thumbnail} alt={post.title} /> }
        <div className="card-body">
          <div className={`card-title ${ this.props.type === 'small_background' && this.props.type === 'small_image' && this.props.type === 'mini' && 'card-title-small'}`}><a href={post.link}>{post.title}</a></div>
          { this.props.type === 'largest' && <p className="card-text">{post.description}</p> }
          { this.props.type != 'mini' && <small className="post_meta"><a href="#">{post.author}</a><span>{post.date_published}</span></small> }
        </div>
      </div>
    )
  }
}
