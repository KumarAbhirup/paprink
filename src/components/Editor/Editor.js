import React, { Component } from 'react'
import Head from 'next/head'
import dynamic from 'next/dynamic'
import { convertToHTML as stateToHTML } from 'draft-convert'
import HtmlToReact from 'html-to-react'

const Dante = dynamic(import('Dante2'), {
  ssr: false
})

import { ImageBlockConfig } from 'Dante2/package/lib/components/blocks/image'
import { EmbedBlockConfig } from 'Dante2/package/lib/components/blocks/embed'
import { VideoBlockConfig } from 'Dante2/package/lib/components/blocks/video'
import { PlaceholderBlockConfig } from 'Dante2/package/lib/components/blocks/placeholder'

export default class PaprinkEditor extends Component {

  state = {}

  shouldComponentUpdate() {
    return true
  }

  displayHtml() {
    var Parser = HtmlToReact.Parser
    var htmlToReactParser = new Parser()
    return htmlToReactParser.parse(this.state.editorHtml)
  }

  defaultContent = this.props.editorContent ? this.props.editorContent : {"blocks":[{"key":"duei8","text":"This is Your Editor","type":"header-one","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"8i1c7","text":"Use it to your advantage.","type":"header-two","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"7vs85","text":"PaprInk, where writers meet opportunities.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}

  render() {
    return (
      <>
      <Dante
        content={this.defaultContent}
        body_placeholder={'Write your next masterpiece ✍️'}
        onChange={async editor => {
          // All these editor states should be saved to the database for future use
          await this.setState({ editorCurrentContent: editor.state.editorState._immutable.currentContent }) // this is what we use to convert into html
          await this.setState({ editorSerializedOutput: editor.emitSerializedOutput() }) // object that looks like this.defaultContent
          await this.props.editorState({editorSerializedOutput: this.state.editorSerializedOutput, editorCurrentContent: this.state.editorCurrentContent})
        }}
        widgets={[
          ImageBlockConfig({
              options: {
                  upload_handler: async (file, imageBlock) => {

                    const data = new FormData()
                    data.append('file', file)
                    data.append('upload_preset', 'paprinkEditor')

                    await fetch(`https://api.cloudinary.com/v1_1/${process.env.CLOUDINARY_USERNAME}/image/upload`, {
                      method: 'POST',
                      body: data
                    })
                    .then(async res => {
                      const payload = await res.json()
                      imageBlock.uploadCompleted(payload.secure_url)
                    })
                    .catch(err => {
                      imageBlock.uploadFailed()
                    })

                  }
              }
          }),
          VideoBlockConfig({ options: { placeholder: 'Paste the link of a YouTube or Vimeo video!', endpoint: '//noembed.com/embed?url=', caption: 'optional caption' } }), 
          EmbedBlockConfig({ options: { placeholder: 'Embed a tweet, or a YouTube video.', endpoint: '//noembed.com/embed?url=' } }),
          PlaceholderBlockConfig()
        ]}
        read_only={0}
        ref="editor"
      />
      </>
    )
  }

}