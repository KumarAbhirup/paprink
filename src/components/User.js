import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import PropTypes from 'prop-types'

export const CURRENT_USER_QUERY = gql`
  query {
    me {
      id
      name
      fname
      lname
      email
      previledge
      gender
      birthday
      profilePicture
      signUpMethod
    }
  }
`

export const getMe = async client => {
  const me = await client.query({
    query: CURRENT_USER_QUERY
  })
  return me
}

const User = props => (
  <Query {...props} query={CURRENT_USER_QUERY}>
    {payload => {
      return props.children(payload)
    }}
  </Query>
)

User.propTypes = {
  children: PropTypes.func.isRequired,
}

export default User