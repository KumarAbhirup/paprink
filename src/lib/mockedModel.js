import casual from 'casual'

const editorSerializedOutput = '{"blocks":[{"key":"duei8","text":"This is Your Editor","type":"header-one","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"8i1c7","text":"Use it to your advantage.","type":"header-two","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"7vs85","text":"PaprInk, where writers meet opportunities.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}'
const editorCurrentContent = '{"test": "test"}'
const thumbnail = '{"image":"https://res.cloudinary.com/iqubex/image/upload/v1559537917/paprink/fp4kdl4wgxtelfhualpy.jpg","uploading":"done","smallImage":"https://res.cloudinary.com/iqubex/image/upload/c_scale,q_auto,w_730/v1559537917/paprink/fp4kdl4wgxtelfhualpy.jpg","blackOverlayImage":"https://res.cloudinary.com/iqubex/image/upload/b_rgb:000000,c_scale,o_57,q_auto,w_1032,z_1.2/v1559537917/paprink/fp4kdl4wgxtelfhualpy.jpg","smallCardImage":"https://res.cloudinary.com/iqubex/image/upload/c_fill,f_jpg,g_center,h_174,w_263/v1559537917/paprink/fp4kdl4wgxtelfhualpy.jpg"}'
const profilePicture = "https://res.cloudinary.com/iqubex/image/upload/c_crop,g_face,h_700,w_700/v1559462578/paprink/n5qnngxyxmzzhiupsw7l.jpg"
import { dogImage } from '../tests/Editor/editor.test'

// seed it so we get consistent results
casual.seed(777)

const date = new Date()

const fakeCategory = () => ({
  __typename: 'Category',
  id: casual.word,
  text: casual.word,
  category: casual.word.toUpperCase(),
  posts: []
})

const fakeUser = () => ({
  __typename: 'User',
  id: casual.uuid,
  socialId: 'social123',
  fname: casual.first_name,
  lname: casual.last_name,
  username: casual.username,
  name: casual.name,
  phone: casual.phone,
  email: casual.email,
  gender: null,
  birthday: null,
  upvotes: [],
  bio: casual.description,
  // posts: [fakePost()],
  profilePicture,
  followers: [],
  previledge: ['AUTHOR'],
  signUpMethod: 'test',
  accessToken: casual.uuid
})

const fakePost = () => ({
  __typename: 'Post',
  id: casual.uuid,
  author: fakeUser(),
  title: casual.sentence,
  thumbnail: {image: dogImage},
  categories: [fakeCategory()],
  status: "PUBLISHED",
  slug: "this-is-a-title-i-love",
  editorSerializedOutput: {blocks: [{test: 'test'}]},
  editorCurrentContent: {test: 'test'},
  editorHtml: null,
  publishedAt: date.toISOString(),
  upvotes: [],
  upvotesNumber: 0,
  authorId: fakeUser().id,
  refUrl: null
})

// Fake LocalStorage
class LocalStorageMock {
  constructor() {
    this.store = {}
  }

  clear() {
    this.store = {}
  }

  getItem(key) {
    return this.store[key] || null
  }

  setItem(key, value) {
    this.store[key] = value.toString()
  }

  removeItem(key) {
    delete this.store[key]
  }
}

export {
  LocalStorageMock,
  fakeUser,
  fakePost,
  fakeCategory
}