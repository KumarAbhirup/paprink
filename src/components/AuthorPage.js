import React, { Component } from 'react'
import { Grid } from 'styled-css-grid'

import PageContent from './PageContent'
import Sidebar from './Sidebar'
import Card from './Card/'

export default class AuthorPage extends Component {
  render() {
    return (
			<>
      <PageContent>

        {/* Page Content */}

				<div className="col-lg-9">
					<div className="main_content" style={{paddingBottom: "40px"}}>

						<div className="col-md-6" style={{position: "relative", margin: "20px auto", color: "#ffffff"}}>
							<div className="material-card Pink" style={{width: "350px"}}>
								<h2>
										<span><b>{ this.props.authorData.name }</b></span>
										<strong>
												<i className="fa fa-fw fa-at"></i>
												{ this.props.authorData.username }
										</strong>
								</h2>
								<div className="mc-content">
										<div className="img-container">
												<img className="img-responsive" src={this.props.authorData.profilePicture} style={{display: "block", maxWidth: "100%", height: "100%"}} />
										</div>
										<div className="mc-description" style={{color: "black", marginTop: "-10px"}}>
												{ this.props.authorData.bio || "He hasn't yet written anything about him! ðŸ˜…" }
										</div>
								</div>
								<a className="mc-btn-action">
										<i className="fa fa-bars"></i>
								</a>
							</div>
						</div>

						<div className="category">
							<div className="section_content" style={{width: "100%"}}>
								<div style={{maxWidth: "900px", width: "100%"}}>
								<Grid
									columns="repeat(auto-fill, 260px)"
									gap="20px"
                >
									{ this.props.posts.length > 0 ? this.props.posts.map(post => {
										return <Card type={'small_image'} post={post} key={post.id} author user={this.props.user} />
									}) : <p>No posts available.</p> }
								</Grid>
								</div>
							</div>
						</div>

					{ this.props.pageInfo.hasNextPage && (
						<div className="load_more">
							<div id="load_more" className="load_more_button text-center trans_200" onClick={this.props.onLoadMore}>Load More</div>
						</div>
					) }

					</div>
				</div>

        {/* End of Page Content */}

        <Sidebar />

      </PageContent>
			</>
    )
  }
}
