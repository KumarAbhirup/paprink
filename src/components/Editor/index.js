import React, { Component } from 'react'
import styled from 'styled-components'
import Router, { withRouter } from 'next/router'
import { ApolloConsumer } from 'react-apollo'
import { Button as BootstrapButton } from 'react-bootstrap'
import isEmpty from 'lodash.isempty'
import gql from 'graphql-tag'
import PageContent from '../PageContent'
import CategorySelector from './CategorySelector'
import Editor from './Editor'
import ImageUploader from './ImageUploader'

const TitleInputBox = styled.input`
  width: 100%;
  border: none;
  padding: 30px;
  font-size: 28px;
  border-radius: 6px;
  &:focus {
    outline: none;
  }
`

const SAVE_POST_MUTATION = gql`
  mutation SAVE_POST_MUTATION(
    $title: String!
    $editorSerializedOutput: Json!
    $editorCurrentContent: Json!
    $editorHtml: String!
    $categories: [Category]!
    $thumbnail: Json!
    $status: PostStatus!
  ) {
    savePost(
      title: $title
      editorSerializedOutput: $editorSerializedOutput
      editorCurrentContent: $editorCurrentContent
      editorHtml: $editorHtml
      categories: $categories
      thumbnail: $thumbnail
      status: $status
    ) {
      id
      author {
        id
        email
        name
        fname
        lname
      }
      title
      thumbnail
      categories
      status
    }
  }
`

class EditorPage extends Component {

  state = {
    title: 'Write an awesome title!',
    // editorContent: {},
		categories: [],
    images: {},
    error: false,
    published: false
  }

  onTitleChange = async event => {
    await this.setState({ title: event.target.value })
		this.props.titleState(this.state.title)
  }

  publish = async client => {
    await this.setState({ error: false })
    if (this.state.title.length === 0 || this.state.categories.length === 0 || isEmpty(this.state.images) || this.state.images.image === null || this.state.editorSerializedOutput.blocks.length === 1) {
      await this.setState({ error: true })
    } else {
      await this.setState({ published: 'loading' })
      var savePost = await client.mutate({
        mutation: SAVE_POST_MUTATION,
        variables: {
          title: this.state.title,
          thumbnail: this.state.images,
          editorHtml: this.state.editorHtml,
          editorSerializedOutput: this.state.editorSerializedOutput,
          editorCurrentContent: this.state.editorCurrentContent,
          categories: this.state.categories.map(object => (object.id.toUpperCase())),
          status: 'PUBLISHED'
        }
      })
      if (savePost.data.savePost){
        await this.setState({ published: true })
        this.props.router.push('/editor?postId=10', '/editor/10', { shallow: true })
      } else {
        await this.setState({ published: 'error' })
      }
    }
  }

  render() {
    return (
      <PageContent noSidebar>

        {/* Post Content */}

        <ApolloConsumer>
          {client => (
            
            <div className="col-lg-10 offset-lg-1">
              <div className="post_content">

                <TitleInputBox type="text" placeholder="Write an awesome title!" value={this.state.title === '' ? null : this.state.title} onChange={event => this.onTitleChange(event)} maxLength="55" />

                <hr style={{opacity: 0.3}} />

                <div className="post_body" style={{marginTop: "20px", color: "#000000"}}>

                  <div className="post_tags" style={{margin: '10px auto'}}>
                    <ul>
                      { this.state.categories.map(category => <li className="post_tag"><a href="#">{category.text}</a></li>) }
                    </ul>
                  </div>

                  <Editor editorState={async editorContent => await this.setState({ editorSerializedOutput: editorContent.editorSerializedOutput, editorCurrentContent: editorContent.editorCurrentContent, editorHtml: editorContent.editorHtml })} />

                </div>

                <CategorySelector categoryState={async categories => {
                  await this.setState({ categories })
                  this.props.categoryState(this.state.categories)
                }} />

                <ImageUploader imageState={async images => {
                  await this.setState({ images })
                  this.props.imageState(this.state.images)
                }} />
                
                <div className="post_panel bottom_panel d-flex flex-row align-items-center justify-content-end">
                  { this.state.error && <p style={{color: "red", fontWeight: "bold"}}>You have to fill all those blanks!</p> }
                  &nbsp; &nbsp; &nbsp;
                  <BootstrapButton variant="dark" style={{marginRight: "10px", cursor: 'pointer'}}>{this.state.published ? '📝 SAVE CHANGES' :'📝 SAVE AS DRAFT'}</BootstrapButton>
                  {!this.state.published && <BootstrapButton variant={this.state.published === 'error' ? "danger" : this.state.published === true ? "info" : "success"} style={{cursor: 'pointer'}} onClick={() => this.publish(client)}>{this.state.published === true ? '👌 UPDATE' : this.state.published === 'error' ? 'Something went wrong ☹️' : this.state.published === 'loading' ? 'PUBLISHING...' : '🎉 PUBLISH'}</BootstrapButton>}
                </div>

                <div className="post_panel bottom_panel d-flex flex-row align-items-center justify-content-center">
                  Thanks for spending your time here. ❤️ from PaprInk Team!
                </div>

              </div>
            </div>
          )}
        </ApolloConsumer>

        {/* End of Post Content */}
        
      </PageContent>
    )
  }

}

export default withRouter(EditorPage)