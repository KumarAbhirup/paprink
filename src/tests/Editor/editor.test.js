import { mount } from 'enzyme'
import { MockedProvider } from 'react-apollo/test-utils'
import Router from 'next/router'
import toJson from 'enzyme-to-json'
import wait from 'waait'

import Editor from '../../components/Editor'
import { CURRENT_USER_QUERY } from '../../components/User'
import { fakeUser, fakePost } from '../../lib/mockedModel'

Router.router = { push: jest.fn() }

// Mock loadable component
const LoadableComponent = ({ children }) => children
jest.mock('next-server/dynamic', () => () => 'Dante')

describe('Editor from scratch', () => {
  const mocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: {
        data: {
          me: fakeUser()
        }
      }
    }
  ]

  it('it Renders properly', async () => {
    const wrapper = await mount(
      <MockedProvider mocks={mocks}>
        <Editor titleState={title => {}} new />
      </MockedProvider>
    )
    expect(toJson(wrapper.find('[data-test="EditorContent"]'))).toMatchSnapshot()
  })

  it('it updates title', async () => {
    const wrapper = await mount(
      <MockedProvider mocks={mocks}>
        <Editor titleState={title => {}} new />
      </MockedProvider>
    )
    const titleBox = wrapper.find('input[data-test="TitleInputBox"]')
    const component = wrapper.find('EditorPage').instance()
    // console.log(wrapper.find('input[data-test="TitleInputBox"]').debug())
    titleBox.simulate('change', { target: { value: "Random title" } })
    // wrapper.update()
    await wait()
    expect(component.state.title).toEqual("Random title")
  })
})