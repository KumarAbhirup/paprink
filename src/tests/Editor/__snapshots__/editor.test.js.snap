// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor from scratch it Renders properly 1`] = `
<div
  className="post_content"
  data-test="EditorContent"
>
  <styled.input
    data-test="TitleInputBox"
    maxLength="55"
    onChange={[Function]}
    placeholder="Write an awesome title!"
    type="text"
    value="Write an awesome title!"
  >
    <StyledComponent
      data-test="TitleInputBox"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "kKiSZs",
            "rules": Array [
              "
  width: 100%;
  border: none;
  padding: 30px;
  font-size: 28px;
  border-radius: 6px;
  &:focus {
    outline: none;
  }
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      maxLength="55"
      onChange={[Function]}
      placeholder="Write an awesome title!"
      type="text"
      value="Write an awesome title!"
    >
      <input
        className="sc-bdVaJa kKiSZs"
        data-test="TitleInputBox"
        maxLength="55"
        onChange={[Function]}
        placeholder="Write an awesome title!"
        type="text"
        value="Write an awesome title!"
      />
    </StyledComponent>
  </styled.input>
  <hr
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <div
    className="post_body"
    style={
      Object {
        "color": "#000000",
        "marginTop": "20px",
      }
    }
  >
    <div
      className="post_tags"
      style={
        Object {
          "margin": "10px auto",
        }
      }
    >
      <ul />
    </div>
    <PaprinkEditor
      editorState={[Function]}
    >
      <Dante
        body_placeholder="Write your next masterpiece ✍️"
        content={
          Object {
            "blocks": Array [
              Object {
                "data": Object {},
                "depth": 0,
                "entityRanges": Array [],
                "inlineStyleRanges": Array [],
                "key": "duei8",
                "text": "This is Your Editor",
                "type": "header-one",
              },
              Object {
                "data": Object {},
                "depth": 0,
                "entityRanges": Array [],
                "inlineStyleRanges": Array [],
                "key": "8i1c7",
                "text": "Use it to your advantage.",
                "type": "header-two",
              },
              Object {
                "data": Object {},
                "depth": 0,
                "entityRanges": Array [],
                "inlineStyleRanges": Array [],
                "key": "7vs85",
                "text": "PaprInk, where writers meet opportunities.",
                "type": "unstyled",
              },
            ],
            "entityMap": Object {},
          }
        }
        onChange={[Function]}
        read_only={0}
        widgets={
          Array [
            Object {
              "block": [Function],
              "breakOnContinuous": true,
              "editable": true,
              "handleEnterWithText": [Function],
              "handleEnterWithoutText": [Function],
              "icon": [Function],
              "options": Object {
                "upload_handler": [Function],
              },
              "renderable": true,
              "selectedFn": [Function],
              "selected_class": "is-selected is-mediaFocused",
              "title": "add an image",
              "type": "image",
              "widget_options": Object {
                "displayOnInlineTooltip": true,
                "insert_block": "image",
                "insertion": "upload",
              },
              "wrapper_class": "graf graf--figure",
            },
            Object {
              "block": [Function],
              "breakOnContinuous": true,
              "editable": true,
              "handleEnterWithText": [Function],
              "handleEnterWithoutText": [Function],
              "icon": [Function],
              "options": Object {
                "caption": "optional caption",
                "endpoint": "//noembed.com/embed?url=",
                "placeholder": "Paste the link of a YouTube or Vimeo video!",
              },
              "renderable": true,
              "selected_class": " is-selected is-mediaFocused",
              "title": "insert video",
              "type": "video",
              "widget_options": Object {
                "displayOnInlineTooltip": true,
                "insert_block": "video",
                "insertion": "placeholder",
              },
              "wrapper_class": "graf--figure graf--iframe",
            },
            Object {
              "block": [Function],
              "breakOnContinuous": true,
              "editable": true,
              "handleEnterWithText": [Function],
              "handleEnterWithoutText": [Function],
              "icon": [Function],
              "options": Object {
                "endpoint": "//noembed.com/embed?url=",
                "placeholder": "Embed a tweet, or a YouTube video.",
              },
              "renderable": true,
              "selected_class": "is-selected is-mediaFocused",
              "title": "insert embed",
              "type": "embed",
              "widget_options": Object {
                "displayOnInlineTooltip": true,
                "insert_block": "embed",
                "insertion": "placeholder",
              },
              "wrapper_class": "graf graf--mixtapeEmbed",
            },
            Object {
              "block": [Function],
              "breakOnContinuous": true,
              "editable": true,
              "handleEnterWithText": [Function],
              "handleEnterWithoutText": [Function],
              "renderable": true,
              "selected_class": "is-selected is-mediaFocused",
              "type": "placeholder",
              "widget_options": Object {
                "displayOnInlineTooltip": false,
              },
              "wrapper_class": "is-embedable",
            },
          ]
        }
      />
    </PaprinkEditor>
  </div>
  <CategorySelector
    categories={Array []}
    categoryState={[Function]}
  >
    <Head>
      <SideEffect
        handleStateChange={null}
        isAmp={false}
        reduceComponentsToState={[Function]}
      />
    </Head>
    <div
      className="white-box-avision"
    >
      <h3
        style={
          Object {
            "marginLeft": "3px",
          }
        }
      >
        🏗️ Category
      </h3>
      <DragDropContext(ReactTags)
        allowDeleteFromEmptyInput={false}
        allowDragDrop={false}
        delimiters={
          Array [
            188,
            13,
          ]
        }
        handleAddition={[Function]}
        handleDelete={[Function]}
        handleDrag={[Function]}
        maxLength={24}
        minQueryLength={1}
        placeholder="Add at least 1 Category (max 3)"
        suggestions={
          Array [
            Object {
              "id": "tech",
              "text": "Technology",
            },
            Object {
              "id": "fin",
              "text": "Finance",
            },
            Object {
              "id": "digimark",
              "text": "Digital Marketing",
            },
            Object {
              "id": "coding",
              "text": "Programming",
            },
            Object {
              "id": "tutorial",
              "text": "Tutorial",
            },
            Object {
              "id": "howto",
              "text": "How To",
            },
            Object {
              "id": "writing",
              "text": "Writing",
            },
            Object {
              "id": "inspire",
              "text": "Inspirational",
            },
            Object {
              "id": "science",
              "text": "Science",
            },
            Object {
              "id": "politics",
              "text": "Politics",
            },
            Object {
              "id": "lifestyle",
              "text": "Lifestyle",
            },
            Object {
              "id": "food",
              "text": "Food",
            },
            Object {
              "id": "bussiness",
              "text": "Bussiness",
            },
            Object {
              "id": "entrepreneur",
              "text": "Entrepreneurs",
            },
            Object {
              "id": "history",
              "text": "History",
            },
            Object {
              "id": "health",
              "text": "Health",
            },
            Object {
              "id": "pet",
              "text": "Pets",
            },
            Object {
              "id": "parenthood",
              "text": "Parenthood",
            },
            Object {
              "id": "travel",
              "text": "Travel",
            },
            Object {
              "id": "india",
              "text": "India",
            },
            Object {
              "id": "china",
              "text": "China",
            },
            Object {
              "id": "uk",
              "text": "United Kingdom",
            },
            Object {
              "id": "us",
              "text": "United States",
            },
            Object {
              "id": "world",
              "text": "World",
            },
            Object {
              "id": "news",
              "text": "News",
            },
            Object {
              "id": "review",
              "text": "Product Review",
            },
            Object {
              "id": "art",
              "text": "Art",
            },
            Object {
              "id": "culture",
              "text": "Culture",
            },
            Object {
              "id": "story",
              "text": "Story",
            },
          ]
        }
        tags={Array []}
      >
        <ReactTags
          allowAdditionFromPaste={true}
          allowDeleteFromEmptyInput={false}
          allowDragDrop={false}
          allowUnique={true}
          autocomplete={false}
          autofocus={true}
          delimiters={
            Array [
              188,
              13,
            ]
          }
          handleAddition={[Function]}
          handleDelete={[Function]}
          handleDrag={[Function]}
          inline={true}
          inputFieldPosition="inline"
          labelField="text"
          maxLength={24}
          minQueryLength={1}
          placeholder="Add at least 1 Category (max 3)"
          readOnly={false}
          resetInputOnDelete={true}
          suggestions={
            Array [
              Object {
                "id": "tech",
                "text": "Technology",
              },
              Object {
                "id": "fin",
                "text": "Finance",
              },
              Object {
                "id": "digimark",
                "text": "Digital Marketing",
              },
              Object {
                "id": "coding",
                "text": "Programming",
              },
              Object {
                "id": "tutorial",
                "text": "Tutorial",
              },
              Object {
                "id": "howto",
                "text": "How To",
              },
              Object {
                "id": "writing",
                "text": "Writing",
              },
              Object {
                "id": "inspire",
                "text": "Inspirational",
              },
              Object {
                "id": "science",
                "text": "Science",
              },
              Object {
                "id": "politics",
                "text": "Politics",
              },
              Object {
                "id": "lifestyle",
                "text": "Lifestyle",
              },
              Object {
                "id": "food",
                "text": "Food",
              },
              Object {
                "id": "bussiness",
                "text": "Bussiness",
              },
              Object {
                "id": "entrepreneur",
                "text": "Entrepreneurs",
              },
              Object {
                "id": "history",
                "text": "History",
              },
              Object {
                "id": "health",
                "text": "Health",
              },
              Object {
                "id": "pet",
                "text": "Pets",
              },
              Object {
                "id": "parenthood",
                "text": "Parenthood",
              },
              Object {
                "id": "travel",
                "text": "Travel",
              },
              Object {
                "id": "india",
                "text": "India",
              },
              Object {
                "id": "china",
                "text": "China",
              },
              Object {
                "id": "uk",
                "text": "United Kingdom",
              },
              Object {
                "id": "us",
                "text": "United States",
              },
              Object {
                "id": "world",
                "text": "World",
              },
              Object {
                "id": "news",
                "text": "News",
              },
              Object {
                "id": "review",
                "text": "Product Review",
              },
              Object {
                "id": "art",
                "text": "Art",
              },
              Object {
                "id": "culture",
                "text": "Culture",
              },
              Object {
                "id": "story",
                "text": "Story",
              },
            ]
          }
          tags={Array []}
        >
          <div
            className="ReactTags__tags react-tags-wrapper"
          >
            <div
              className="ReactTags__selected"
            >
              <div
                className="ReactTags__tagInput"
              >
                <input
                  aria-label="Add at least 1 Category (max 3)"
                  className="ReactTags__tagInputField"
                  maxLength={24}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onPaste={[Function]}
                  placeholder="Add at least 1 Category (max 3)"
                  type="text"
                />
                <Suggestions
                  classNames={
                    Object {
                      "activeSuggestion": "ReactTags__activeSuggestion",
                      "remove": "ReactTags__remove",
                      "selected": "ReactTags__selected",
                      "suggestions": "ReactTags__suggestions",
                      "tag": "ReactTags__tag",
                      "tagInput": "ReactTags__tagInput",
                      "tagInputField": "ReactTags__tagInputField",
                      "tags": "ReactTags__tags",
                    }
                  }
                  handleClick={[Function]}
                  handleHover={[Function]}
                  isFocused={false}
                  labelField="text"
                  minQueryLength={1}
                  query=""
                  selectedIndex={-1}
                  suggestions={
                    Array [
                      Object {
                        "id": "tech",
                        "text": "Technology",
                      },
                      Object {
                        "id": "fin",
                        "text": "Finance",
                      },
                      Object {
                        "id": "digimark",
                        "text": "Digital Marketing",
                      },
                      Object {
                        "id": "coding",
                        "text": "Programming",
                      },
                      Object {
                        "id": "tutorial",
                        "text": "Tutorial",
                      },
                      Object {
                        "id": "howto",
                        "text": "How To",
                      },
                      Object {
                        "id": "writing",
                        "text": "Writing",
                      },
                      Object {
                        "id": "inspire",
                        "text": "Inspirational",
                      },
                      Object {
                        "id": "science",
                        "text": "Science",
                      },
                      Object {
                        "id": "politics",
                        "text": "Politics",
                      },
                      Object {
                        "id": "lifestyle",
                        "text": "Lifestyle",
                      },
                      Object {
                        "id": "food",
                        "text": "Food",
                      },
                      Object {
                        "id": "bussiness",
                        "text": "Bussiness",
                      },
                      Object {
                        "id": "entrepreneur",
                        "text": "Entrepreneurs",
                      },
                      Object {
                        "id": "history",
                        "text": "History",
                      },
                      Object {
                        "id": "health",
                        "text": "Health",
                      },
                      Object {
                        "id": "pet",
                        "text": "Pets",
                      },
                      Object {
                        "id": "parenthood",
                        "text": "Parenthood",
                      },
                      Object {
                        "id": "travel",
                        "text": "Travel",
                      },
                      Object {
                        "id": "india",
                        "text": "India",
                      },
                      Object {
                        "id": "china",
                        "text": "China",
                      },
                      Object {
                        "id": "uk",
                        "text": "United Kingdom",
                      },
                      Object {
                        "id": "us",
                        "text": "United States",
                      },
                      Object {
                        "id": "world",
                        "text": "World",
                      },
                      Object {
                        "id": "news",
                        "text": "News",
                      },
                      Object {
                        "id": "review",
                        "text": "Product Review",
                      },
                      Object {
                        "id": "art",
                        "text": "Art",
                      },
                      Object {
                        "id": "culture",
                        "text": "Culture",
                      },
                      Object {
                        "id": "story",
                        "text": "Story",
                      },
                    ]
                  }
                />
              </div>
            </div>
          </div>
        </ReactTags>
      </DragDropContext(ReactTags)>
    </div>
  </CategorySelector>
  <ImageUploader
    imageState={[Function]}
    uploadPreset="paprink"
  >
    <div
      className="white-box-avision"
    >
      <h3>
        🖼️ Relevant Thumbnail
      </h3>
      <div
        className="bottom_panel d-flex flex-row align-items-center justify-content-start"
        style={
          Object {
            "padding": "15px 0px",
          }
        }
      >
           
        <input
          id="thumbnailUpload"
          name="thumbnailUpload"
          onChange={[Function]}
          placeholder="Upload the article thumbnail."
          style={
            Object {
              "width": "100%",
            }
          }
          type="file"
        />
           
      </div>
    </div>
  </ImageUploader>
  <div
    className="post_panel bottom_panel d-flex flex-row align-items-center justify-content-end"
  >
         
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "marginRight": "10px",
        }
      }
      type="button"
      variant="dark"
    >
      <button
        className="btn btn-dark"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "marginRight": "10px",
          }
        }
        type="button"
      >
        📝 SAVE AS DRAFT
      </button>
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
      type="button"
      variant="success"
    >
      <button
        className="btn btn-success"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
        type="button"
      >
        🎉 PUBLISH
      </button>
    </Button>
  </div>
  <div
    className="post_panel bottom_panel d-flex flex-row align-items-center justify-content-center"
  >
    Thanks for spending your time here. ❤️ from PaprInk Team!
  </div>
</div>
`;
